app/scripts/controllers/service.js

```diff
@@ -27,11 +29,48 @@ angular.module('openshiftConsole')
       }
     ];
 
+    $scope.podFailureReasons = {
+      "Pending": "This pod will not receive traffic until all of its containers have been created."
+    };
+
     var watches = [];
 
+    // receives routes for the current service and maps a service port to each route name
+    var getPortsForRoutes = function(service, routes) {
+      if (!service || !routes) {
+        return;
+      }
+
+      $scope.portsForRoutes = {};
+      var orderedPorts = $filter("orderBy")(service.spec.ports, "port");
+
+      for(var i = 0; i < routes.length; i++) {
+        $scope.portsForRoutes[routes[i].metadata.name] = orderedPorts[i];
+
+        // determine if at least one nodePort present
+        if(orderedPorts[i].nodePort) {
+          $scope.showNodePorts = true;
+        }
+      }
+    };
+
+    // receive pods for the current service scope only when the service object is available
+    var getPodsForService = function(service, pods) {
+      $scope.podsForService = {};
+      if (!service || !pods) {
+        return;
+      }
+
+      var ls = new LabelSelector(service.spec.selector);
+      $scope.podsForService = ls.select(pods);
+    };
     var serviceResolved = function(service, action) {
       $scope.loaded = true;
       $scope.service = service;
+
+      getPodsForService(service, $scope.pods);
```

Comment:
service and pods are already on the scope, you don't need to pass them into the functions